root/artist/ArtistSubscribers.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import {l, ln} from '../static/scripts/common/i18n';
+import EditorLink from '../static/scripts/common/components/EditorLink';
+
+import ArtistLayout from './ArtistLayout';
+
+type Props = {|
+  +canonicalURL: string,
+  +entity: ArtistT,
+  +privateEditors: number,
+  +publicEditors: $ReadOnlyArray<EditorT>,
+  +subscribed: boolean,
+|};
+
+const ArtistSubscribers = ({
+  canonicalURL,
+  entity,
+  privateEditors,
+  publicEditors,
+  subscribed,
+}: Props) => (
+  <ArtistLayout artist={entity} canonicalURL={canonicalURL} page="subscribers" title={l('Subscribers')}>
+    <h2>{l('Subscribers')}</h2>
+
+    {(publicEditors.length || (privateEditors > 0)) ? (
+      <>
+        <p>{ln(
+          'There is currently {num} user subscribed to {artist}:',
+          'There are currently {num} users subscribed to {artist}:',
```

Comment:
Would it be possible to have a single, generic `root/entity/Subscribers.js` component using the `chooseLayoutComponent` utility you added?

From what I can tell, only some of the strings differ, and that's no issue if we use `N_ln` etc. and store them as constants at the top of the file:

```JavaScript
const subscriberCountMessages = {
  artist: N_ln(
    'There is currently {num} user subscribed to {artist}:',
    'There are currently {num} users subscribed to {artist}:',
  ),
  label: N_ln(/* ... */),
};

// below, in render()...
<p>
  {subscriberCountMessages[entityType].toLocaleString(
    count,
    {[entityType]: entity.name, num: count},
  )}
</p>
```

Reply:
The URL should always be the same except for the entity type, so `` `/account/subscriptions/${entityType}/remove?id=${entity.id}` `` should work without needing a constant. AFAICT there are only two translated strings that would need to be moved to use `N_ln` — for the rest we could just use JS template strings and computed property names (e.g. `{[entityType]: foo}` is the same as `{artist: foo}` if `entityType` is "artist").